hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkDataSourceDAGExecution.scala

```diff
@@ -89,7 +89,7 @@ class TestSparkDataSourceDAGExecution extends HoodieSparkClientTestBase with Sca
   @CsvSource(Array(
     "upsert,org.apache.hudi.client.SparkRDDWriteClient.commit",
     "insert,org.apache.hudi.client.SparkRDDWriteClient.commit",
-    "bulk_insert,org.apache.hudi.HoodieSparkSqlWriter$.bulkInsertAsRow"))
+    "bulk_insert,org.apache.hudi.HoodieDatasetBulkInsertHelper.bulkInsert"))
```

Comment:
`bulk_insert` with `Overwrite` mode will change to use `DatasetBulkInsertOverwriteTableCommitActionExecutor` to avoid deleting basePath first. So it would call `FSUtils.getAllPartitionPaths` to get all existing partition paths, which would introduce extra stages containing `HoodieSparkSqlWriter$.bulkInsertAsRow`, but it won't call write operation multi times, can see from the screenshot.

![Screenshot 2023-05-23 at 14 45 58](https://github.com/apache/hudi/assets/10115332/d2e0cf3b-65ed-45a3-b786-0778f9b8e395)

Reply:
Got it.